src/openforms/js/components/admin/form_design/registrations/objectsapi/AddressNlObjectsApiVariableConfigurationEditor.js

```diff
@@ -0,0 +1,347 @@
+import {REGISTRATION_OBJECTS_TARGET_PATHS} from 'components/admin/form_design/constants';
+import Field from 'components/admin/forms/Field';
+import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+import {Checkbox} from 'components/admin/forms/Inputs';
+import Select, {LOADING_OPTION} from 'components/admin/forms/Select';
+import ErrorMessage from 'components/errors/ErrorMessage';
+import {post} from 'utils/fetch';
+
+const ADDRESSNL_NESTED_PROPERTIES = {
+  postcodeTargetPath: {type: 'string'},
+  houseLetterTargetPath: {type: 'string'},
+  houseNumberTargetPath: {type: 'string'},
+  houseNumberAdditionTargetPath: {type: 'string'},
+  cityTargetPath: {type: 'string'},
+  streetNameTargetPath: {type: 'string'},
+};
+
+const asJsonSchema = isSpecificTargetPaths => {
+  // Special handling for addressNL component type.
+  if (!isSpecificTargetPaths) {
+    return {
+      type: 'object',
+      properties: ADDRESSNL_NESTED_PROPERTIES,
+    };
+  } else {
+    return {
+      type: 'string',
+    };
+  }
+};
+
+export const AddressNlEditor = ({
+  variable,
+  components,
+  namePrefix,
+  isGeometry,
+  index,
+  mappedVariable,
+  objecttype,
+  objectsApiGroup,
+  objecttypeVersion,
+  TargetPathDisplay,
+}) => {
+  const {csrftoken} = useContext(APIContext);
+  const {values, setValues, setFieldValue} = useFormikContext();
+  const [jsonSchemaVisible, toggleJsonSchemaVisible] = useToggle(false);
+  const {specificTargetPaths} = values;
+  const [isSpecificTargetPaths, setIsSpecificTargetPaths] = useState(specificTargetPaths || false);
+
+  const deriveAddress = components[variable?.key]['deriveAddress'];
+
+  const {
+    loading,
+    value: targetPaths,
+    error,
+  } = useAsync(async () => {
+    const response = await post(REGISTRATION_OBJECTS_TARGET_PATHS, csrftoken, {
+      objectsApiGroup,
+      objecttype,
+      objecttypeVersion,
+      variableJsonSchema: asJsonSchema(isSpecificTargetPaths),
+    });
+    if (!response.ok) {
+      throw new Error('Error when loading target paths');
+    }
+
+    return response.data;
+  }, [isSpecificTargetPaths]);
+
+  const getTargetPath = pathSegment => targetPaths.find(t => isEqual(t.targetPath, pathSegment));
+
+  const choices =
+    loading || error
+      ? LOADING_OPTION
+      : targetPaths.map(t => [JSON.stringify(t.targetPath), <TargetPathDisplay target={t} />]);
+
+  if (error)
+    return (
+      <ErrorMessage>
+        <FormattedMessage
+          description="Objects API variable registration configuration API error"
+          defaultMessage="Something went wrong when fetching the available target paths"
+        />
+      </ErrorMessage>
+    );
+
+  const onSpecificTargetPathsChange = event => {
+    setIsSpecificTargetPaths(event.target.checked);
+
+    // update Formik's state in order to be able to have this value as the default one for the React's state
```

Comment:
I don't think we need the `isSpecificTargetPaths` when you also update the `specificTargetPaths`. Updating `specificTargetPaths` should trigger a reload, that updates the UI, so `isSpecificTargetPaths` is probably obsolete.

Reply:
So i think you can just remove the state, and use `specificTargetPaths` for the checks